File: backend/src/main/java/com/bytechat/services/MeetingService.java
package com.bytechat.services;

import com.bytechat.dto.response.MeetingResponse;
import com.bytechat.entity.User;

import java.util.List;

public interface MeetingService {
    MeetingResponse createMeeting(Long channelId, String title, String password, User currentUser);
    List<MeetingResponse> getActiveWorkspaceMeetings(Long workspaceId, User currentUser);
    MeetingResponse joinMeeting(Long meetingId, String password, User currentUser);
    void endMeeting(Long meetingId, User currentUser);
}

File: frontend/src/services/meetingService.js
import api from './api';

const meetingService = {
  createMeeting: (channelId, title, password) =>
    api.post(`/meetings/channels/${channelId}`, { title, password }),
  getWorkspaceMeetings: (workspaceId) => api.get(`/meetings/workspaces/${workspaceId}`),
  getMeeting: (meetingId) => api.get(`/meetings/${meetingId}`),
  joinMeeting: (meetingId, password) => api.post(`/meetings/${meetingId}/join`, { password }),
  endMeeting: (meetingId) => api.post(`/meetings/${meetingId}/end`),
};

export default meetingService;
